src/Icicle/Avalanche/Statement/Simp/Trample.hs

```diff
@@ -0,0 +1,185 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternGuards #-}
+{-# LANGUAGE PatternSynonyms #-}
+
+module Icicle.Avalanche.Statement.Simp.Trample (
+    trample
+  ) where
+
+import           Icicle.Avalanche.Prim.Flat
+import           Icicle.Avalanche.Statement.Simp
+import           Icicle.Avalanche.Statement.Statement
+import qualified Icicle.Common.Exp.Prim.Minimal as Min
+
+import           Icicle.Common.Base
+import           Icicle.Common.Exp
+import           Icicle.Common.Fresh
+import           Icicle.Common.Type
+
+import           P
+
+import qualified Data.Map as Map
+
+
+------------------------------------------------------------------------
+
+pattern PrimZip  ta tb = PrimArray   (PrimArrayZip ta tb)
+pattern PrimPair ta tb = PrimMinimal (Min.PrimConst (Min.PrimConstPair ta tb))
+pattern PrimFst  ta tb = PrimMinimal (Min.PrimPair  (Min.PrimPairFst   ta tb))
+pattern PrimSnd  ta tb = PrimMinimal (Min.PrimPair  (Min.PrimPairSnd   ta tb))
+pattern PrimMkOpt   tv = PrimOption  (PrimOptionPack          tv)
+pattern PrimIsSome  tv = PrimProject (PrimProjectOptionIsSome tv)
+pattern PrimGet     tv = PrimUnsafe  (PrimUnsafeOptionGet     tv)
+
+
+------------------------------------------------------------------------
+
+trample :: (Show n, Ord n)
+        => a
+        -> Statement a n Prim
+        -> Fresh n (Statement a n Prim)
```

Comment:
yep, so melt currently does this for pairs already, trample is the same but works for options too